shared/trieutil/sparse_merkle.go

```diff
@@ -1,31 +1,35 @@
 package trieutil
 
 import (
+	"bytes"
 	"errors"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 
 	"github.com/prysmaticlabs/prysm/shared/hashutil"
 )
 
 // MerkleTrie implements a sparse, general purpose Merkle trie to be used
 // across ETH2.0 Phase 0 functionality.
 type MerkleTrie struct {
-	branches [][][32]byte
+	branches [][][]byte
+	originalItems [][]byte // list of provided items before hashing them into leaves.
```

Comment:
Since we are mostly dealing with protos, we change this to use rawbytes instead of [32]byte otherwise we would be doing painful conversions each time.